UDPServer/SiceUDPServer.cs

```diff
@@ -38,17 +44,24 @@ public void Connect()
                 token = source.Token;
                 Task.Run(() => Listen(), token);
                 this.ConnectionStatus = true;
-                eventAggregator.PublishOnUIThread(new ServerStatusChangedEvent(true));
+                eventAggregator.PublishOnUIThread(new ServerStatusChangedEvent(ServerStatus.Connected));
             }
         }
 
         private async Task Listen()
         {
             while (true)
             {
-                if (DataAvailable != 0)
+                if (ConnectionStatus && DataAvailable != 0)
                 {
-                    await GetData();
+                    if(HasLoggedIn)
+                    {
+                        await GetData();
+                    }
+                    else
+                    {
+                        await GetLogin();
+                    }
```

Comment:
This shouldn't work this way. What if there are multiple clients logging into this server? If the first client has logged in, then the other client's won't.

What should happen is that it should always listen for data and the protocol should indicate what kind of information it is.